jobs/job_utils.py

```diff
@@ -158,7 +175,170 @@ def get_model_from_beam_entity(beam_entity):
     Returns:
         datastore_services.Model. The NDB model representation of the entity.
     """
-    model_id = get_model_id(beam_entity)
-    model_class = (
-        datastore_services.Model._lookup_model(get_model_kind(beam_entity))) # pylint: disable=protected-access
+    beam_key = beam_entity.key.to_client_key()
+    model_id = beam_key.id_or_name
+    model_class = datastore_services.Model._lookup_model(beam_key.kind) # pylint: disable=protected-access
     return model_class(id=model_id, **beam_entity.properties)
+
+
+def get_beam_query_from_ndb_query(query):
+    """Returns an equivalent Apache Beam query from the given NDB query.
```

Comment:
Should we explain here why we need these methods? I imagine getting to this code year from now without any context it might be confusing why exactly is this code needed.

Reply:
Done in #12890: https://github.com/oppia/oppia/pull/12890#discussion_r638281238